R/create_package.R

```diff
@@ -16,9 +16,12 @@
 #' package <- create_package()
 #' str(package)
 create_package <- function() {
-  list(
-    profile = "tabular-data-package",
-    resources = list(),
-    directory = "." # Current directory
+  structure(
```

Comment:
There is old code that used to add a `datapackage` class:

https://github.com/frictionlessdata/frictionless-r/blob/01e383f81272ba32fe33ca933896e1e5a7a5f77f/R/create_package.R#L26-L28

It first creates the list, and only then assigns the class (no `structure()`). It also stores any already existing classes, rather than hardcoding `list`. Can you use that approach instead?

Reply:
Yup! Also reworked create_package a bit so that it creates from a descriptor, so we can then use it as a factory in read_package